src/Sylius/Bundle/PayumBundle/Action/ConvertPaymentToPaypalProAction.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Lakion package.
+ *
+ * (c) Lakion
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\PayumBundle\Action;
+
+use Payum\Core\Action\ActionInterface;
+use Payum\Core\Exception\RequestNotSupportedException;
+use Payum\Core\Request\Convert;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\PaymentInterface;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class ConvertPaymentToPaypalProAction implements ActionInterface
```

Comment:
Not sure this class is needed, this one already does same thing: https://github.com/Payum/Payum/blob/master/src/Payum/Paypal/ProCheckout/Nvp/Action/ConvertPaymentAction.php#L12

Doesn't it work for you?